arangod/Cluster/Maintenance/Action.cpp

```diff
@@ -22,79 +22,85 @@
 /// @author Matthew Von-Maszewski
 ////////////////////////////////////////////////////////////////////////////////
 
-#include "Action.h"
-
-#include "Cluster/CreateCollection.h"
-#include "Cluster/CreateDatabase.h"
-#include "Cluster/DropCollection.h"
-#include "Cluster/DropDatabase.h"
-#include "Cluster/DropIndex.h"
-#include "Cluster/EnsureIndex.h"
-#include "Cluster/MaintenanceStrings.h"
-#include "Cluster/NonAction.h"
-#include "Cluster/ResignShardLeadership.h"
-#include "Cluster/SynchronizeShard.h"
-#include "Cluster/TakeoverShardLeadership.h"
-#include "Cluster/UpdateCollection.h"
+#include "Cluster/Maintenance/Action.h"
+
+#include "Cluster/Maintenance/Actions/CreateCollection.h"
+#include "Cluster/Maintenance/Actions/CreateDatabase.h"
+#include "Cluster/Maintenance/Actions/DropCollection.h"
+#include "Cluster/Maintenance/Actions/DropDatabase.h"
+#include "Cluster/Maintenance/Actions/DropIndex.h"
+#include "Cluster/Maintenance/Actions/EnsureIndex.h"
+#include "Cluster/Maintenance/Actions/NonAction.h"
+#include "Cluster/Maintenance/Actions/ResignShardLeadership.h"
+#include "Cluster/Maintenance/Actions/SynchronizeShard.h"
+#include "Cluster/Maintenance/Actions/TakeoverShardLeadership.h"
+#include "Cluster/Maintenance/Actions/UpdateCollection.h"
+#include "Cluster/Maintenance/Actions/UpgradeCollection.h"
+#include "Cluster/Maintenance/MaintenanceStrings.h"
 
 #include "Logger/Logger.h"
 
 using namespace arangodb;
 using namespace arangodb::maintenance;
 
 using factories_t =
-    std::unordered_map<std::string, std::function<std::unique_ptr<ActionBase>(MaintenanceFeature&, ActionDescription const&)>>;
+    std::unordered_map<std::string, std::function<std::shared_ptr<ActionBase>(MaintenanceFeature&, ActionDescription const&)>>;
 
```

Comment:
Again, my mistake probably. Missed. why we went to shared pointers here. This is technically a factory.